assets/report_template.Rmd

```diff
@@ -0,0 +1,1450 @@
+if ( params$dada_sample_inference == "independent" ) {
+    cat("ASVs were inferred for each sample independently.")
+} else if ( params$dada_sample_inference == "pooled" ) {
+    cat("ASVs were inferred from pooled sample information.")
+} else {
+    cat("ASVs were initally inferred for each sample independently, but re-examined with all samples (pseudo-pooled).")
+}
+```
+
+```{r, results='asis'}
+flag_any_filtering <- !isFALSE(params$path_barrnap_sum) || !isFALSE(params$filter_len_asv) || !isFALSE(params$filter_codons)
+```
+
+<!-- Section on ASV filtering -->
+
+```{r, eval = flag_any_filtering, results='asis'}
+cat("# Filtering of ASVs\n")
+```
+
+<!-- Subsection on rRNA classification with Barrnap -->
+
+```{r, eval = !isFALSE(params$path_barrnap_sum), results='asis'}
+cat("## rRNA detection\n")
+cat("[Barrnap](https://github.com/tseemann/barrnap) classifies the ASVs into the origin domain (including mitochondiral origin).\n\n", sep = "")
+
+# Read the barrnap files and count the lines
+barrnap_sum = read.table( params$path_barrnap_sum, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
+# keep only ASV_ID & eval columns & sort
+barrnap_sum <- subset(barrnap_sum, select = c(ASV_ID,mito_eval,euk_eval,arc_eval,bac_eval))
+# choose kingdom (column) with lowest evalue
+barrnap_sum[is.na(barrnap_sum)] <- 1
+barrnap_sum$result = colnames(barrnap_sum[,2:5])[apply(barrnap_sum[,2:5],1,which.min)]
+barrnap_sum$result = gsub("_eval", "", barrnap_sum$result)
+
+#import asv table
+asv_table <- readLines(params$path_asv_fa)
+n_asv <- sum(grepl("^>", asv_table))
+
+# calculate numbers
+n_classified <- length(barrnap_sum$result)
+n_bac <- sum(grepl("bac", barrnap_sum$result))
+n_arc <- sum(grepl("arc", barrnap_sum$result))
+n_mito <- sum(grepl("mito", barrnap_sum$result))
+n_euk <- sum(grepl("euk", barrnap_sum$result))
+
+barrnap_df_sum <- data.frame(label=c('Bacteria','Archaea','Mitochondria','Eukaryotes','Unclassified'),
+                 count=c(n_bac,n_arc,n_mito,n_euk,n_asv - n_classified),
+                 percent=c(round( (n_bac/n_asv)*100, 2), round( (n_arc/n_asv)*100, 2), round( (n_mito/n_asv)*100, 2), round( (n_euk/n_asv)*100, 2), round( ( (n_asv - n_classified) /n_asv)*100, 2) ) )
+
+# Build outputtext
+cat( "Barrnap classified ")
+cat( barrnap_df_sum$count[1], "(", barrnap_df_sum$percent[1],"%) ASVs as most similar to Bacteria, " )
+cat( barrnap_df_sum$count[2], "(", barrnap_df_sum$percent[2],"%) ASVs to Archea, " )
+cat( barrnap_df_sum$count[3], "(", barrnap_df_sum$percent[3],"%) ASVs to Mitochondria, " )
+cat( barrnap_df_sum$count[4], "(", barrnap_df_sum$percent[4],"%) ASVs to Eukaryotes, and " )
+cat( barrnap_df_sum$count[5], "(", barrnap_df_sum$percent[5],"%) were below similarity threshold to any kingdom." )
+
+# Barplot
+plot_barrnap_df_sum <- ggplot(barrnap_df_sum,
+        aes(x = reorder(label, desc(label)), y = percent)) +
+        geom_bar(stat = "identity", fill = rgb(0.1, 0.4, 0.75), width = 0.5) +
```

Comment:
Maybe we could add a fixed `ylim(0, 100)`?

Reply:
yes that looks nice, done